lib/grpc/logger.ex

```diff
@@ -22,6 +22,7 @@ defmodule GRPC.Logger.Server do
 
   def call(req, stream, next, level) do
     if Logger.compare_levels(level, Logger.level()) != :lt do
+      Logger.metadata(request_id: stream.request_id)
```

Comment:
I would only override the key if it's not present. It's a really common key and as it is done here, can be nil-ified if `stream.request_id` is `nil`
```suggestion
      Logger.metadata(request_id: Logger.metadata()[:request_id] || stream.request_id)
```